perses/tests/test_relative.py

```diff
@@ -577,3 +577,273 @@ def test_HybridTopologyFactory_energies(molecule_perturbation_list = [['naphthal
+        subtracted_valence_energy = 0.0
+    else:
+        subtracted_valence_energy = geometry_engine.reverse_final_context_reduced_potential - geometry_engine.reverse_atoms_with_positions_reduced_potential
+
+    # Generate repartitioned htf at lambda = 0
+    zero_htf = RepartitionedHybridTopologyFactory(topology_proposal=topology_proposal,
+                          current_positions=positions,
+                          new_positions=new_positions,
+                          endstate=0)
+
+    # Compute error at lambda = 0 endstate
+    zero_state_error, _ = validate_endstate_energies(zero_htf._topology_proposal,
+                                                     zero_htf,
+                                                     added_valence_energy,
+                                                     subtracted_valence_energy,
+                                                     ENERGY_THRESHOLD=ENERGY_THRESHOLD,
+                                                     platform=openmm.Platform.getPlatformByName('Reference'),
+                                                     repartitioned_endstate=0)
+    # Generate repartitioned htf at lambda = 1
+    one_htf = RepartitionedHybridTopologyFactory(topology_proposal=topology_proposal,
+                                                  current_positions=positions,
+                                                  new_positions=new_positions,
+                                                  endstate=1)
+
+    # Compute error at lambda = 1 endstate
+    _, one_state_error = validate_endstate_energies(one_htf._topology_proposal,
+                                                     one_htf,
+                                                     added_valence_energy,
+                                                     subtracted_valence_energy,
+                                                     ENERGY_THRESHOLD=ENERGY_THRESHOLD,
+                                                     platform=openmm.Platform.getPlatformByName('Reference'),
+                                                     repartitioned_endstate=1)
+
+    # Check that endstate errors are below threshold
+    assert abs(zero_state_error) < ENERGY_THRESHOLD, f"The zero state alchemical and nonalchemical energy absolute difference {abs(zero_state_error)} is greater than the threshold of {ENERGY_THRESHOLD}."
+    assert abs(one_state_error) < ENERGY_THRESHOLD, f"The one state alchemical and nonalchemical energy absolute difference {abs(one_state_error)} is greater than the threshold of {ENERGY_THRESHOLD}."
+
+    print(f"Abs difference in zero alchemical vs nonalchemical systems: {abs(zero_state_error)}")
+    print(f"Abs difference in one alchemical vs nonalchemical systems: {abs(one_state_error)}")
+
+
+def test_RepartitionedHybridTopologyFactory_energies():
+    """
+    Test whether the difference in the nonalchemical zero and alchemical zero states is the forward valence energy.  Also test for the one states.
```

Comment:
probbably  better idea to clarify and distinguish between these test_..._energies since they are subtle but important